### PR TITLE
Pistes improvements

### LINKED ISSUE #6391: Ski piste areas rendered wrong
Organic Maps
Screenshot_20231024-225713

OsmAnd
Screenshot_20231024-230015

OpenSnowMap
Screenshot_20231024-230135

### PR DESCRIPTION
**Pull request by pastk**
- add `piste:type=hike/skitour/connection/snow_park`
- color freeride pistes in yellow-orange, consider ungroomed downhill runs as freeride
- Outdoors:
  - increase pistes visibilities to z12-
  - add areal fills from z13-

- pull names from `piste:name`
- support `ref=` and pull from `piste:ref`

Closes #6391

### CODE REVIEW

**Comment by pastk** [comment]
Does it look too promiment for the main style?
Maybe would be better to leave it mostly as is for the main (no area fills, thin lines) and apply the new looks to Outdoors only?

image

@vng @biodranik

**Comment by biodranik** [comment]
Seeing non-existing ski trails in summer is not a good idea. Should we move them to a separate winter style/layer instead?

**Comment by pastk** [comment]
> Seeing non-existing ski trails in summer is not a good idea. Should we move them to a separate winter style/layer instead?

When we have that winter style - for sure.

Even in summer its good to know there was a ski run. At least you can be sure there are no trees :) And overall the ground could be more artificially smoothed compared to the surroundings.

**Comment by pastk** [comment]
@organicmaps/translations please review new piste types names here: https://github.com/organicmaps/organicmaps/pull/6987/commits/d64c63037066deefd4cb114b0da4dbc4cef7ec95

**Comment by d4f5409d** [comment]
> Seeing non-existing ski trails in summer is not a good idea. Should we move them to a separate winter style/layer instead?

That's not a good idea because on the south globe winter is at a different time

**Comment by RedAuburn** [comment]
maybe a generic sports layer

**Comment by pastk** [comment]
> That's not a good idea because on the globe's south winter is at a different time

We don't adjust the map style with each season change if that's what you mean.

Historically the current OM map style is more "summer".
As we don't have a designated "winter/snow" map style yet, we have to compromise and integrate winter-specific features into the map style we have.

**Comment by pastk** [comment]
I left the main style mostly as is; z14:
image

Outdoors has higher visibility of pistes; z12:
image

And displays area fills at higher zooms; e.g. z14:
image

**Comment by biodranik** [comment]
We obviously know if it's a southern or northern part of the globe. It would be great to reduce winter sports visibility to a minimum when there is summer in the area. Are there any stats about average ski slopes opening/closing times?

If winter sports goes to the outdoors ("Active Sports") style, not into the main one, then it's ok.

**Comment by pastk** [comment]
Need to put it through beta or test data gen to test more cases (skitours, refs processing...).

File: data/strings/types_strings.txt [hunk 1]
```diff
@@ -30062,6 +30146,182 @@
...
+    hi = पिस्ट कनेक्शन
+    hu = Pálya kapcsolat
+    id = Koneksi Piste
+    it = Connessione alle piste
+    ja = ピステ・コネクション
+    ko = 피스트 연결
+    lt = "Piste Connection
+    mr = Piste कनेक्शन
+    nb = Pisteforbindelse
+    nl = Piste aansluiting
```
Comment (jimcarst) [inline data/strings/types_strings.txt:30266]:
```suggestion
    nl = Pisteverbinding
```

**Comment by matheusgomesms** [comment]
Maybe there is a mix in commits? There is a translation for poles, trees, footway as well.

File: data/strings/types_strings.txt [hunk 2]
```diff
@@ -30062,6 +30146,182 @@
...
+    pl = Piste Connection
+    pt = Ligação às pistas
+    pt-BR = Conexão de pista
```
Comment (matheusgomesms) [inline data/strings/types_strings.txt:30269]:
```suggestion
    pt-BR = Conexão às pistas
```

File: data/strings/types_strings.txt [hunk 3]
```diff
@@ -30062,6 +30146,182 @@
...
+    id = Jalur Skitour
+    it = Sentiero Skitour
+    ja = スキーツアートレイル
+    ko = 스키투어 트레일
+    lt = "Skitour" takas
+    mr = Skitour ट्रेल
+    nb = Skitour-stien
+    nl = Skitourspoor
+    pl = Szlak Skitour
+    pt = Trilha Skitour
```
Comment (matheusgomesms) [inline data/strings/types_strings.txt:30312]:
```suggestion
    pt = Trilho de esqui de passeio
    pt-BR = Trilha de esqui de passeio
```

**Review by matheusgomesms — CHANGES_REQUESTED** [review]

**Comment by pastk** [comment]
> Maybe there is a mix in commits? There is a translation for poles, trees, footway as well.

Fixed! Thanks for spotting!

**Review by d4f5409d — CHANGES_REQUESTED** [review]
Hu has serious issues and sadly I ain't capable fixing it. I really don't know what we use for those words.

File: data/mapcss-mapping.csv [hunk 4]
```diff
@@ -689,18 +689,18 @@ place|continent;601;
...
-deprecated|deprecated;614;x
-deprecated|deprecated;615;x
-deprecated|deprecated;616;x
+piste:type|downhill|novice|area;[piste:type=downhill][piste:difficulty=novice][area?];;;;609;
+piste:type|downhill|easy|area;[piste:type=downhill][piste:difficulty=easy][area?];;;;610;
+piste:type|downhill|intermediate|area;[piste:type=downhill][piste:difficulty=intermediate][area?];;;;611;
+piste:type|downhill|advanced|area;[piste:type=downhill][piste:difficulty=advanced][area?];;;;612;
+piste:type|downhill|expert|area;[piste:type=downhill][piste:difficulty=expert][area?];;;;613;
+piste:type|downhill|area;[piste:type=downhill][area?];;;;614;
+piste:type|sled|area;[piste:type=sled][area?];;;;615;
```
Comment (vng) [inline data/mapcss-mapping.csv:702]: Do we need separate area types?
I suppose we can define line and area rules for the initial piste:type|downhill|novice type and everything should be ok, no?

Comment (pastk) [reply data/mapcss-mapping.csv:702]: Then areas will have a border around them (doesn't look good IMO, especially when two areas connect to each other there will be a line separating them) and we won't have an option to render just centerlines without areas (like it is now in the main style).

Comment (vng) [reply data/mapcss-mapping.csv:702]: Hm, should we fix it? Like use line rules for line objects only, and area rules for area objects only?

We don't draw _the center line_ for area objects anyway.

Comment (vng) [reply data/mapcss-mapping.csv:702]: I don't like _generating_ new types. We crutch drawing but lose in some other processing (like routing or search).

Comment (pastk) [reply data/mapcss-mapping.csv:702]: This should be discussed and implemented separately in conjunction with #5884 and other cases like dams and some barriers..

Comment (pastk) [reply data/mapcss-mapping.csv:702]: > I don't like _generating_ new types. We crutch drawing but lose in some other processing (like routing or search).

This particular PR doesn't impact search or routing anyhow, right?

Comment (pastk) [reply data/mapcss-mapping.csv:702]: https://github.com/organicmaps/organicmaps/issues/7075

**Review by vng — APPROVED** [review]

**Pull request merged**